src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +58,42 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
+    }
+
+    /**
+     * Customization of org.apache.maven.doxia.sink.impl.XhtmlBaseSink.unifyEOL() method.
+     *
+     * @param text the text to scan.
+     * @return a String that contains only Linux System EOLs.
+     */
+    private static String unifyToLinuxEndOfLine(String text) {
+        return text
+                .replace("\r\n", LINUX_EOL)
+                .replace('\r', LINUX_EOL.charAt(0));
+    }
+
+    /**
+     * Overriden from org.apache.maven.doxia.sink.impl.Xhtml5BaseSink class.
+     * Ommited the usage of tableContentWriterStack, tableCaptionWriterStack and
+     * tableCaptionXMLWriterStack lists, since they are redundant in our current
+     * context and causing too many CI failures which includes PMD, Spotbugs and
+     * circleci - run inspections.
+     *
+     * @param text the text to write to the destination.
```

Comment:
```suggestion
     * Writes text to a file, converting any line breaks to newlines.
     *
     * @param text the text to write to the destination.
```

Details about CI and classes we don't use are not very valuable here, mention of CircleCI and other static analysis is bound to become outdated, and explicit mention of `Overriden` is redundant, we have `Override` annotation on method already.

Reply:
@kalpadiptyaroy do not resolve items from reviewers